<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>pbmc3k â€¢ sneezy</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="pbmc3k">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">sneezy</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/sneezy.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/pbmc3k.html">pbmc3k</a>
    </li>
    <li>
      <a href="../articles/pdfsense.html">pdfsense: or understanding high dimensional parameter space</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/sa-lee/sneezy">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>pbmc3k</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/sa-lee/sneezy/blob/master/vignettes/pbmc3k.Rmd"><code>vignettes/pbmc3k.Rmd</code></a></small>
      <div class="hidden name"><code>pbmc3k.Rmd</code></div>

    </div>

    
    
<p>Here we explore the use of <code>sneezy</code> on the 10x genomics PBMC3k single-cell dataset. We follow the preprocessing and quality control steps from <a href="https://osca.bioconductor.org/pbmc-3k-10x-dataset-filtered.html">Chapter 25 of Orchestrating Single Cell Analysis with R</a>, and use that to explore the linear and non-linear embedding spaces via tours and diagnostics.</p>
<div id="setup" class="section level2">
<h2 class="hasAnchor">
<a href="#setup" class="anchor"></a>Setup</h2>
<p>First, we load up the raw data from the <code>TENxPBMCData</code> package, and the analysis packages <code>scater</code> and <code>scran</code>:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(TENxPBMCData)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="co">#&gt; Loading required package: SingleCellExperiment</span></a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="co">#&gt; Loading required package: SummarizedExperiment</span></a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="co">#&gt; Loading required package: GenomicRanges</span></a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="co">#&gt; Loading required package: stats4</span></a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="co">#&gt; Loading required package: BiocGenerics</span></a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="co">#&gt; Loading required package: parallel</span></a>
<a class="sourceLine" id="cb1-8" data-line-number="8"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-9" data-line-number="9"><span class="co">#&gt; Attaching package: 'BiocGenerics'</span></a>
<a class="sourceLine" id="cb1-10" data-line-number="10"><span class="co">#&gt; The following objects are masked from 'package:parallel':</span></a>
<a class="sourceLine" id="cb1-11" data-line-number="11"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-12" data-line-number="12"><span class="co">#&gt;     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,</span></a>
<a class="sourceLine" id="cb1-13" data-line-number="13"><span class="co">#&gt;     clusterExport, clusterMap, parApply, parCapply, parLapply,</span></a>
<a class="sourceLine" id="cb1-14" data-line-number="14"><span class="co">#&gt;     parLapplyLB, parRapply, parSapply, parSapplyLB</span></a>
<a class="sourceLine" id="cb1-15" data-line-number="15"><span class="co">#&gt; The following objects are masked from 'package:stats':</span></a>
<a class="sourceLine" id="cb1-16" data-line-number="16"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-17" data-line-number="17"><span class="co">#&gt;     IQR, mad, sd, var, xtabs</span></a>
<a class="sourceLine" id="cb1-18" data-line-number="18"><span class="co">#&gt; The following objects are masked from 'package:base':</span></a>
<a class="sourceLine" id="cb1-19" data-line-number="19"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-20" data-line-number="20"><span class="co">#&gt;     anyDuplicated, append, as.data.frame, basename, cbind,</span></a>
<a class="sourceLine" id="cb1-21" data-line-number="21"><span class="co">#&gt;     colnames, dirname, do.call, duplicated, eval, evalq, Filter,</span></a>
<a class="sourceLine" id="cb1-22" data-line-number="22"><span class="co">#&gt;     Find, get, grep, grepl, intersect, is.unsorted, lapply, Map,</span></a>
<a class="sourceLine" id="cb1-23" data-line-number="23"><span class="co">#&gt;     mapply, match, mget, order, paste, pmax, pmax.int, pmin,</span></a>
<a class="sourceLine" id="cb1-24" data-line-number="24"><span class="co">#&gt;     pmin.int, Position, rank, rbind, Reduce, rownames, sapply,</span></a>
<a class="sourceLine" id="cb1-25" data-line-number="25"><span class="co">#&gt;     setdiff, sort, table, tapply, union, unique, unsplit, which,</span></a>
<a class="sourceLine" id="cb1-26" data-line-number="26"><span class="co">#&gt;     which.max, which.min</span></a>
<a class="sourceLine" id="cb1-27" data-line-number="27"><span class="co">#&gt; Loading required package: S4Vectors</span></a>
<a class="sourceLine" id="cb1-28" data-line-number="28"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-29" data-line-number="29"><span class="co">#&gt; Attaching package: 'S4Vectors'</span></a>
<a class="sourceLine" id="cb1-30" data-line-number="30"><span class="co">#&gt; The following object is masked from 'package:base':</span></a>
<a class="sourceLine" id="cb1-31" data-line-number="31"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-32" data-line-number="32"><span class="co">#&gt;     expand.grid</span></a>
<a class="sourceLine" id="cb1-33" data-line-number="33"><span class="co">#&gt; Loading required package: IRanges</span></a>
<a class="sourceLine" id="cb1-34" data-line-number="34"><span class="co">#&gt; Loading required package: GenomeInfoDb</span></a>
<a class="sourceLine" id="cb1-35" data-line-number="35"><span class="co">#&gt; Loading required package: Biobase</span></a>
<a class="sourceLine" id="cb1-36" data-line-number="36"><span class="co">#&gt; Welcome to Bioconductor</span></a>
<a class="sourceLine" id="cb1-37" data-line-number="37"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-38" data-line-number="38"><span class="co">#&gt;     Vignettes contain introductory material; view with</span></a>
<a class="sourceLine" id="cb1-39" data-line-number="39"><span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span></a>
<a class="sourceLine" id="cb1-40" data-line-number="40"><span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></a>
<a class="sourceLine" id="cb1-41" data-line-number="41"><span class="co">#&gt; Loading required package: DelayedArray</span></a>
<a class="sourceLine" id="cb1-42" data-line-number="42"><span class="co">#&gt; Loading required package: matrixStats</span></a>
<a class="sourceLine" id="cb1-43" data-line-number="43"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-44" data-line-number="44"><span class="co">#&gt; Attaching package: 'matrixStats'</span></a>
<a class="sourceLine" id="cb1-45" data-line-number="45"><span class="co">#&gt; The following objects are masked from 'package:Biobase':</span></a>
<a class="sourceLine" id="cb1-46" data-line-number="46"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-47" data-line-number="47"><span class="co">#&gt;     anyMissing, rowMedians</span></a>
<a class="sourceLine" id="cb1-48" data-line-number="48"><span class="co">#&gt; Loading required package: BiocParallel</span></a>
<a class="sourceLine" id="cb1-49" data-line-number="49"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-50" data-line-number="50"><span class="co">#&gt; Attaching package: 'DelayedArray'</span></a>
<a class="sourceLine" id="cb1-51" data-line-number="51"><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></a>
<a class="sourceLine" id="cb1-52" data-line-number="52"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-53" data-line-number="53"><span class="co">#&gt;     colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges</span></a>
<a class="sourceLine" id="cb1-54" data-line-number="54"><span class="co">#&gt; The following objects are masked from 'package:base':</span></a>
<a class="sourceLine" id="cb1-55" data-line-number="55"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-56" data-line-number="56"><span class="co">#&gt;     aperm, apply, rowsum</span></a>
<a class="sourceLine" id="cb1-57" data-line-number="57"><span class="co">#&gt; Loading required package: HDF5Array</span></a>
<a class="sourceLine" id="cb1-58" data-line-number="58"><span class="co">#&gt; Loading required package: rhdf5</span></a>
<a class="sourceLine" id="cb1-59" data-line-number="59"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(scater)</a>
<a class="sourceLine" id="cb1-60" data-line-number="60"><span class="co">#&gt; Loading required package: ggplot2</span></a>
<a class="sourceLine" id="cb1-61" data-line-number="61"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(scran)</a>
<a class="sourceLine" id="cb1-62" data-line-number="62"></a>
<a class="sourceLine" id="cb1-63" data-line-number="63">pbmc3k &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/TENxPBMCData/man/TENxPBMCData.html">TENxPBMCData</a></span>(<span class="st">"pbmc3k"</span>)</a>
<a class="sourceLine" id="cb1-64" data-line-number="64"><span class="co">#&gt; Using temporary cache /tmp/RtmpQIaH6L/BiocFileCache</span></a>
<a class="sourceLine" id="cb1-65" data-line-number="65"><span class="co">#&gt; snapshotDate(): 2019-10-22</span></a>
<a class="sourceLine" id="cb1-66" data-line-number="66"><span class="co">#&gt; Using temporary cache /tmp/RtmpQIaH6L/BiocFileCache</span></a>
<a class="sourceLine" id="cb1-67" data-line-number="67"><span class="co">#&gt; Using temporary cache /tmp/RtmpQIaH6L/BiocFileCache</span></a>
<a class="sourceLine" id="cb1-68" data-line-number="68"><span class="co">#&gt; see ?TENxPBMCData and browseVignettes('TENxPBMCData') for documentation</span></a>
<a class="sourceLine" id="cb1-69" data-line-number="69"><span class="co">#&gt; Using temporary cache /tmp/RtmpQIaH6L/BiocFileCache</span></a>
<a class="sourceLine" id="cb1-70" data-line-number="70"><span class="co">#&gt; Using temporary cache /tmp/RtmpQIaH6L/BiocFileCache</span></a>
<a class="sourceLine" id="cb1-71" data-line-number="71"><span class="co">#&gt; downloading 1 resources</span></a>
<a class="sourceLine" id="cb1-72" data-line-number="72"><span class="co">#&gt; retrieving 1 resource</span></a>
<a class="sourceLine" id="cb1-73" data-line-number="73"><span class="co">#&gt; Using temporary cache /tmp/RtmpQIaH6L/BiocFileCache</span></a>
<a class="sourceLine" id="cb1-74" data-line-number="74"><span class="co">#&gt; loading from cache</span></a>
<a class="sourceLine" id="cb1-75" data-line-number="75"><span class="co">#&gt; Using temporary cache /tmp/RtmpQIaH6L/BiocFileCache</span></a>
<a class="sourceLine" id="cb1-76" data-line-number="76"><span class="co">#&gt; Using temporary cache /tmp/RtmpQIaH6L/BiocFileCache</span></a>
<a class="sourceLine" id="cb1-77" data-line-number="77"><span class="co">#&gt; Using temporary cache /tmp/RtmpQIaH6L/BiocFileCache</span></a>
<a class="sourceLine" id="cb1-78" data-line-number="78">pbmc3k</a>
<a class="sourceLine" id="cb1-79" data-line-number="79"><span class="co">#&gt; class: SingleCellExperiment </span></a>
<a class="sourceLine" id="cb1-80" data-line-number="80"><span class="co">#&gt; dim: 32738 2700 </span></a>
<a class="sourceLine" id="cb1-81" data-line-number="81"><span class="co">#&gt; metadata(0):</span></a>
<a class="sourceLine" id="cb1-82" data-line-number="82"><span class="co">#&gt; assays(1): counts</span></a>
<a class="sourceLine" id="cb1-83" data-line-number="83"><span class="co">#&gt; rownames(32738): ENSG00000243485 ENSG00000237613 ...</span></a>
<a class="sourceLine" id="cb1-84" data-line-number="84"><span class="co">#&gt;   ENSG00000215616 ENSG00000215611</span></a>
<a class="sourceLine" id="cb1-85" data-line-number="85"><span class="co">#&gt; rowData names(3): ENSEMBL_ID Symbol_TENx Symbol</span></a>
<a class="sourceLine" id="cb1-86" data-line-number="86"><span class="co">#&gt; colnames: NULL</span></a>
<a class="sourceLine" id="cb1-87" data-line-number="87"><span class="co">#&gt; colData names(11): Sample Barcode ... Individual Date_published</span></a>
<a class="sourceLine" id="cb1-88" data-line-number="88"><span class="co">#&gt; reducedDimNames(0):</span></a>
<a class="sourceLine" id="cb1-89" data-line-number="89"><span class="co">#&gt; spikeNames(0):</span></a>
<a class="sourceLine" id="cb1-90" data-line-number="90"><span class="co">#&gt; altExpNames(0):</span></a></code></pre></div>
<p>Next we follow the simple QC steps from the aforementioned vignette:</p>
<div id="quality-control" class="section level3">
<h3 class="hasAnchor">
<a href="#quality-control" class="anchor"></a>Quality control</h3>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">is.mito &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/grep.html">grep</a></span>(<span class="st">"MT"</span>, <span class="kw">rowData</span>(pbmc3k)<span class="op">$</span>Symbol_TENx)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">stats &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/scater/man/perCellQCMetrics.html">perCellQCMetrics</a></span>(pbmc3k, <span class="dt">subsets=</span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="dt">Mito=</span>is.mito))</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">high.mito &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/scater/man/isOutlier.html">isOutlier</a></span>(stats<span class="op">$</span>subsets_Mito_percent, <span class="dt">nmads=</span><span class="dv">3</span>, <span class="dt">type=</span><span class="st">"higher"</span>)</a>
<a class="sourceLine" id="cb2-4" data-line-number="4">pbmc3k &lt;-<span class="st"> </span>pbmc3k[,<span class="op">!</span>high.mito]</a></code></pre></div>
</div>
<div id="normalization" class="section level3">
<h3 class="hasAnchor">
<a href="#normalization" class="anchor"></a>Normalization</h3>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">pbmc3k &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/scater/man/logNormCounts.html">logNormCounts</a></span>(pbmc3k)</a></code></pre></div>
</div>
<div id="variance-modelling" class="section level3">
<h3 class="hasAnchor">
<a href="#variance-modelling" class="anchor"></a>Variance modelling</h3>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">dec3k &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/scran/man/modelGeneVar.html">modelGeneVar</a></span>(pbmc3k)</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">chosen.hvgs &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/which.html">which</a></span>(dec3k<span class="op">$</span>bio <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>)</a></code></pre></div>
</div>
</div>
<div id="the-tourexperiment-object-and-a-first-embedding" class="section level2">
<h2 class="hasAnchor">
<a href="#the-tourexperiment-object-and-a-first-embedding" class="anchor"></a>The TourExperiment object and a first embedding</h2>
<p>We provide a simple interface to constructing a <code>TourExperiment</code> from a SingleCellExperiment object. Once we have this object, we can construct linear embedding using PCA. Since, the underlying object is backed by an HDF5 file we use randomized PCA for speed purposes.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(sneezy)</a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="dv">2099</span>)</a>
<a class="sourceLine" id="cb5-3" data-line-number="3">pbmc3k_te &lt;-<span class="st"> </span><span class="kw"><a href="../reference/TourExperiment-class.html">TourExperiment</a></span>(pbmc3k)</a>
<a class="sourceLine" id="cb5-4" data-line-number="4">pbmc3k_te &lt;-<span class="st"> </span><span class="kw"><a href="../reference/embed_linear.html">embed_linear</a></span>(pbmc3k_te, </a>
<a class="sourceLine" id="cb5-5" data-line-number="5">                          <span class="dt">num_comp =</span> <span class="dv">10</span>, </a>
<a class="sourceLine" id="cb5-6" data-line-number="6">                          <span class="dt">.on =</span> <span class="st">"logcounts"</span>, </a>
<a class="sourceLine" id="cb5-7" data-line-number="7">                          <span class="dt">.subset =</span> chosen.hvgs, </a>
<a class="sourceLine" id="cb5-8" data-line-number="8">                          <span class="dt">.engine =</span> <span class="kw"><a href="../reference/le-backends.html">pca_random</a></span>())</a></code></pre></div>
<p>The arguments above add an embedding to the pbmc3k_te object, using randomized PCA, only on the highly variable genes weâ€™ve constructed above. Since, the procedure is random, we also set a seed.</p>
<p>The usual approach here is to inspect the results of a cluster analysis on top of an embedding, here we have used the standard</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">g &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/scran/man/buildSNNGraph.html">buildSNNGraph</a></span>(pbmc3k_te, <span class="dt">k=</span><span class="dv">15</span>, <span class="dt">use.dimred =</span> <span class="st">'pca_random'</span>)</a>
<a class="sourceLine" id="cb6-2" data-line-number="2">clust &lt;-<span class="st"> </span>igraph<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/igraph/man/cluster_louvain.html">cluster_louvain</a></span>(g)</a>
<a class="sourceLine" id="cb6-3" data-line-number="3">pbmc3k_te<span class="op">$</span>cluster &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(igraph<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/igraph/man/communities.html">membership</a></span>(clust))</a></code></pre></div>
<p>We can also estimate a basis set via a grand tour:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">pbmc3k_te &lt;-<span class="st"> </span><span class="kw"><a href="../reference/generate_bases.html">generate_bases</a></span>(</a>
<a class="sourceLine" id="cb7-2" data-line-number="2">  pbmc3k_te,</a>
<a class="sourceLine" id="cb7-3" data-line-number="3">  <span class="dt">.on =</span> <span class="st">"pca_random"</span>,</a>
<a class="sourceLine" id="cb7-4" data-line-number="4">  <span class="dt">max_bases =</span> <span class="dv">10</span>,</a>
<a class="sourceLine" id="cb7-5" data-line-number="5">  <span class="dt">clamp  =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb7-6" data-line-number="6">  <span class="dt">.engine =</span> tourr<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/tourr/man/grand_tour.html">grand_tour</a></span>(),</a>
<a class="sourceLine" id="cb7-7" data-line-number="7">)</a>
<a class="sourceLine" id="cb7-8" data-line-number="8">pbmc3k_te</a>
<a class="sourceLine" id="cb7-9" data-line-number="9"><span class="co">#&gt; class: TourExperiment </span></a>
<a class="sourceLine" id="cb7-10" data-line-number="10"><span class="co">#&gt; dim: 32738 2609 </span></a>
<a class="sourceLine" id="cb7-11" data-line-number="11"><span class="co">#&gt; metadata(0):</span></a>
<a class="sourceLine" id="cb7-12" data-line-number="12"><span class="co">#&gt; assays(2): counts logcounts</span></a>
<a class="sourceLine" id="cb7-13" data-line-number="13"><span class="co">#&gt; rownames(32738): ENSG00000243485 ENSG00000237613 ...</span></a>
<a class="sourceLine" id="cb7-14" data-line-number="14"><span class="co">#&gt;   ENSG00000215616 ENSG00000215611</span></a>
<a class="sourceLine" id="cb7-15" data-line-number="15"><span class="co">#&gt; rowData names(3): ENSEMBL_ID Symbol_TENx Symbol</span></a>
<a class="sourceLine" id="cb7-16" data-line-number="16"><span class="co">#&gt; colnames: NULL</span></a>
<a class="sourceLine" id="cb7-17" data-line-number="17"><span class="co">#&gt; colData names(12): Sample Barcode ... Date_published cluster</span></a>
<a class="sourceLine" id="cb7-18" data-line-number="18"><span class="co">#&gt; reducedDimNames(1): pca_random</span></a>
<a class="sourceLine" id="cb7-19" data-line-number="19"><span class="co">#&gt; spikeNames(0):</span></a>
<a class="sourceLine" id="cb7-20" data-line-number="20"><span class="co">#&gt; altExpNames(0):</span></a>
<a class="sourceLine" id="cb7-21" data-line-number="21"><span class="co">#&gt; neighborSetNames(0): </span></a>
<a class="sourceLine" id="cb7-22" data-line-number="22"><span class="co">#&gt; basisSetNames(1): pca_random</span></a></code></pre></div>
<p>This creates an array with dimensions 10, 2, 10 named <code>pca_random</code>. To access the underlying array use <code><a href="../reference/basisSets.html">basisSet(pbmc3k_te)</a></code>.</p>
<p>We can then view the principal components using the <code><a href="../reference/view_tour_xy.html">view_tour_xy()</a></code>, this is a shiny application. You can view an example of it here:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw"><a href="../reference/sneezy_tour.html">sneezy_tour</a></span>(pbmc3k_te)</a></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#setup">Setup</a></li>
      <li><a href="#the-tourexperiment-object-and-a-first-embedding">The TourExperiment object and a first embedding</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Stuart Lee.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
